lmos-classifier-llm/src/main/kotlin/org/eclipse/lmos/classifier/llm/DefaultModelAgentClassifier.kt

```diff
@@ -20,21 +28,40 @@ import org.slf4j.LoggerFactory
 class DefaultModelAgentClassifier(
     private val chatModel: ChatModel,
     private val systemPrompt: String,
+    private val objectMapper: ObjectMapper = jacksonObjectMapper(),
 ) : ModelAgentClassifier {
     private val logger = LoggerFactory.getLogger(javaClass)
 
     override fun classify(request: ClassificationRequest): ClassificationResult {
+        val chatRequest = prepareChatRequest(request)
+        val chatResponse = chatModel.chat(chatRequest)
+        val classificationResult = prepareClassificationResult(chatResponse, request.inputContext.agents)
+        logger.info(
+            "[${javaClass.simpleName}] Classified agent '${classificationResult.agents}' for query '${request.inputContext.userMessage}'.",
+        )
+        return classificationResult
+    }
+
+    private fun prepareChatRequest(request: ClassificationRequest): ChatRequest {
+        val responseFormat =
+            ResponseFormat
+                .builder()
+                .type(ResponseFormatType.JSON)
+                .jsonSchema(JsonSchemas.jsonSchemaFrom(LlmResponse::class.java).get())
```

Comment:
Maybe we can give LLMReponse a better name, e.g. ClassifiedAgent or ClassifiedAgents.